### PR TITLE
chore: minor Improvements to providerquerymanager

### PR DESCRIPTION
**Pull request by gammazero**
Each of the first 5 commits is a stand-alone enhancement.

### CODE REVIEW

File: routing/providerquerymanager/providerquerymanager.go [hunk 1]
```diff
@@ -108,7 +108,9 @@ func WithMaxTimeout(timeout time.Duration) Option {
 	}
 }
 
-// WithMaxInProcessRequests is the maximum number of requests that can be processed in parallel
+// WithMaxInProcessRequests is the maximum number of requests that can be
+// processed in parallel. If this is 0, then the number is unlimited. Default
+// is defaultMaxInProcessRequests.
```
Comment (hsanjuan) [inline routing/providerquerymanager/providerquerymanager.go]: Nit:
```suggestion
// is defaultMaxInProcessRequests (16).
```

Comment (hsanjuan) [reply routing/providerquerymanager/providerquerymanager.go]: Or we make those defaults public and document them.

File: routing/providerquerymanager/providerquerymanager.go [hunk 2]
```diff
@@ -435,6 +434,9 @@ func (fpqm *finishedProviderQueryMessage) handle(pqm *ProviderQueryManager) {
 		close(listener)
 	}
 	delete(pqm.inProgressRequestStatuses, fpqm.k)
+	if len(pqm.inProgressRequestStatuses) == 0 {
+		pqm.inProgressRequestStatuses = nil
+	}
```
Comment (hsanjuan) [inline routing/providerquerymanager/providerquerymanager.go:449]: Is this to reclaim memory?

File: routing/providerquerymanager/providerquerymanager.go [hunk 3]
```diff
@@ -216,32 +208,36 @@ func (pqm *ProviderQueryManager) FindProvidersAsync(sessionCtx context.Context,
 }
 
 func (pqm *ProviderQueryManager) receiveProviders(sessionCtx context.Context, k cid.Cid, max int, receivedInProgressRequest inProgressRequest, onCloseFn func()) <-chan peer.AddrInfo {
-	// maintains an unbuffered queue for incoming providers for given request for a given session
-	// essentially, as a provider comes in, for a given CID, we want to immediately broadcast to all
-	// sessions that queried that CID, without worrying about whether the client code is actually
-	// reading from the returned channel -- so that the broadcast never blocks
-	// based on: https://medium.com/capital-one-tech/building-an-unbounded-channel-in-go-789e175cd2cd
+	// maintains an unbuffered queue for incoming providers for given request
+	// for a given session. Eessentially, as a provider comes in, for a given
```
Comment (hsanjuan) [inline routing/providerquerymanager/providerquerymanager.go]:
```suggestion
	// for a given session. Essentially, as a provider comes in, for a given
```

**Review by hsanjuan — APPROVED** [review]
LGTM... although I think the improvements may be not so minor ;)

File: routing/providerquerymanager/providerquerymanager.go (hunk 2 above)
Comment (gammazero) [reply routing/providerquerymanager/providerquerymanager.go:449]: yes

**Pull request merged**